cmd/chartsvc/handler.go

```diff
@@ -200,6 +200,35 @@ func getChartVersionValues(w http.ResponseWriter, req *http.Request, params Para
 	w.Write([]byte(files.Values))
 }
 
+func findLatestChart(name, version, appversion string) []*models.Chart {
+	db, closer := dbSession.DB()
+	defer closer()
+
+	var charts []*models.Chart
+	if err := db.C(chartCollection).Find(bson.M{
+		"name": name,
+		"chartversions": bson.M{
+			"$elemMatch": bson.M{"version": version, "appversion": appversion},
+		}}).All(&charts); err != nil {
+		log.WithError(err).Errorf("could not find the given chart with id %s", name)
+		// continue to return empty list
+	}
+
+	return charts
+}
+
+// resolveRepos returns the list of repos that contains the given chart and the latest version found
+func resolveRepos(w http.ResponseWriter, req *http.Request, params Params) {
+	latests := findLatestChart(params["chartName"], req.FormValue("version"), req.FormValue("appversion"))
+
+	var cvl apiListResponse
+	for _, cv := range latests {
```

Comment:
Are we not using `newChartListResponse` because in this case we want duplicates? If so, can we add a comment above this for-loop to mention this?

Reply:
Nope, I didn't notice that method.